### PR TITLE
[ll] Update performance and change windows [30/..]

### PR DESCRIPTION
**Pull request by msiglreith**
- Update the performance example
- Optimize GL submission and fix command buffer reset
- Windows now take ownership of the raw windows
- Fix rebasing issues

cc #1321

### CODE REVIEW

File: examples/gamma/main.rs [hunk 1]
```diff
@@ -52,9 +52,10 @@ pub fn main() {
         .with_title("Gamma example".to_string())
         .with_dimensions(1024, 768)
         .with_vsync();
-    let window = gfx_window_glutin::build(builder, &events_loop, ColorFormat::get_format(), DepthFormat::get_format());
+    let win = gfx_window_glutin::build(builder, &events_loop, ColorFormat::get_format(), DepthFormat::get_format());
+    let mut window = gfx_window_glutin::Window::new(window);
```
Comment (kvark) [inline examples/gamma/main.rs]: that must be a mistake? there is no `window`

File: examples/gamma/main.rs [hunk 2]
```diff
@@ -114,6 +117,7 @@ pub fn main() {
         encoder.draw(&slice, &pso, &data);
         encoder.synced_flush(&mut graphics_queue, &[&frame_semaphore], &[&draw_semaphore], None);
         swap_chain.present(&mut graphics_queue, &[&draw_semaphore]);
+        graphics_queue.wait_idle();
```
Comment (kvark) [inline examples/gamma/main.rs]: that one is gone, btw

File: examples/performance/main.rs [hunk 3]
```diff
@@ -89,100 +91,143 @@ trait Renderer: Drop {
...
+        let events_loop = glutin::EventsLoop::new();
+        let builder = glutin::WindowBuilder::new()
+            .with_title("Triangle example".to_string())
+            .with_dimensions(1024, 768)
+            .with_vsync();
+        let win = gfx_window_glutin::build(builder, &events_loop, ColorFormat::get_format(), DepthStencil::get_format());
+        let mut window = gfx_window_glutin::Window::new(win);
+        // Acquire surface and adapters
+        let (mut surface, adapters) = window.get_surface_and_adapters();
+        let queue_descs = adapters[0].get_queue_families().iter()
```
Comment (kvark) [inline examples/performance/main.rs]: can we use the new (shorter) API for these?

File: examples/performance/main.rs [hunk 4]
```diff
@@ -89,100 +91,143 @@ trait Renderer: Drop {
...
+                                    .filter(|family| surface.supports_queue(&family) )
+                                    .map(|family| { (family, 1) })
+                                    .collect::<Vec<_>>();
+
+        // Open device (factory and queues)
+        let gfx::Device { mut factory, mut general_queues, mut graphics_queues, .. } = adapters[0].open(&queue_descs);
+        let mut queue = if let Some(queue) = general_queues.pop() {
```
Comment (kvark) [inline examples/performance/main.rs]: what I was thinking is having a wrapper enum (that is either general or graphics queue) that is dereffable to the graphics one. In this case, that giant `if/else` section could be omitted

File: src/backend/gl/src/command.rs [hunk 5]
```diff
@@ -72,9 +72,17 @@ impl DataBuffer {
     }
 }
 
+#[allow(missing_copy_implementations)]
 pub struct SubmitInfo {
-    pub buf: Vec<Command>,
-    pub data: DataBuffer,
+    // Raw pointer optimization:
+    // Command buffers are stored inside the command pools.
```
Comment (kvark) [inline src/backend/gl/src/command.rs]: a little bit scary, but the comment sounds reasonable

File: examples/gamma/main.rs (hunk 2 above)
Comment (msiglreith) [inline examples/gamma/main.rs]: I haven't implemented it for the `ll` branch so far (core, not corell), but it's on my todo!

File: examples/performance/main.rs (hunk 4 above)
Comment (msiglreith) [inline examples/performance/main.rs]: These if/else clauses are really annoying, but I would probably rather try to 'downcast' the queue families upfront so we only get GraphicsQueue.

**Pull request merged**